misc/multihash/src/hashes.rs

```diff
@@ -26,15 +25,19 @@ pub enum Hash {
     Keccak384,
     /// Keccak-512 (64-byte hash size)
     Keccak512,
+    /// BLAKE2b-512 (64-byte hash size)
+    Blake2b512,
     /// Encoding unsupported
-    Blake2b,
+    Blake2b256,
+    /// BLAKE2s-256 (32-byte hash size)
+    Blake2s256,
     /// Encoding unsupported
-    Blake2s,
+    Blake2s128,
 }
 
 impl Hash {
     /// Get the corresponding hash code.
-    pub fn code(&self) -> u8 {
+    pub fn code(&self) -> u16 {
```

Comment:
Wouldn't this be a backwards incompatible change? Why do we need two bytes to encode the hash algo?

Reply:
Yes, this is technically a backwards incompatible change. The problem is that the hash codes for BLAKE2 (and any future algorithm additions to the multihash spec, for that matter) are too large to fit in a single byte. If you leave the data type a `u8`, the Rust compiler will silently allow the literals to overflow and the BLAKE2 multihashes this library produces will be incorrect.

Ideally, this hash code should be wrapped up in a newtype, e.g. `HashCode`, so we can vary the internal data type freely as the spec changes without breaking the public API. Either way, there will be some necessary breakage to this library to properly support BLAKE2.